events/ratelimiting/unit.go

```diff
@@ -0,0 +1,32 @@
+//go:build unit
+// +build unit
+
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ratelimiting
+
+import "k8s.io/utils/clock"
+
+// RateLimiterWithTicker is a RateLimiter that can be configured with a ticker.
+// Used for testing.
+type RateLimiterWithTicker interface {
+	RateLimiter
+	WithTicker(clock.WithTicker)
+}
+
+func (c *coalescing) WithTicker(clock clock.WithTicker) {
+	c.clock = clock
+}
+
+var _ RateLimiterWithTicker = new(coalescing)
```

Comment:
Using `new` causes the allocation of the object which stays around forever

```suggestion
var _ RateLimiterWithTicker = (*coalescing)(nil)
```

Reply:
Done